server/src/main/java/org/opensearch/tracing/Telemetry.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * Interface for Telemetry providers
+ */
+public interface Telemetry extends Closeable {
+
+    Span createSpan(String spanName, Span parentSpan, Level level);
+
+    Span extractSpanFromHeader(Map<String, String> header);
+
+    BiConsumer<Map<String, String>, Map<String, Object>> injectSpanInHeader();
+
+    void close();
+
```

Comment:
Are these the only abstraction or do we want to add more later
Ref : https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/OpenTelemetry.java

Reply:
We are using Otel for two reasons besides the span lifecycle management(handled by delegating calls to Otel span in OTelSpan.java)
1. For starting the spans (first method)
2. Generating/retrieving tracing context for inter-node communication (second and third method). 

Hence we have added these in the abstraction. We could add other tracing related capabilities of Otel if needed in future. Further we could rename this interface as tracing specific (maybe TracingTelemetry) to clearly call out that this tracing related. Opentelemetry could have other components like metrics, metering etc. Your thoughts?